.github/workflows/main.yaml

```diff
@@ -0,0 +1,312 @@
+      run: |  
+        if [[ -n "${REGISTRY_USERNAME}" ]]; then
+          docker login -u  $REGISTRY_USERNAME -p $REGISTRY_PASSWORD $BASE_SOURCE_IMAGE_REGISTRY
+        fi
+        BASE_IMAGE="${BASE_SOURCE_IMAGE_REGISTRY}${BASE_SOURCE_IMAGE_NAME}:${BASE_SOURCE_IMAGE_TAG}"
+        sed -i -e "s;FROM .*;FROM ${BASE_IMAGE};g" "$file"
+        docker pull $BASE_IMAGE
+      env:
+        context: ${{ matrix.SOURCE_CONTEXT_DIR }}
+        BASE_SOURCE_IMAGE_REGISTRY: ${{ matrix.BASE_SOURCE_IMAGE_REGISTRY }}
+        BASE_SOURCE_IMAGE_NAME: ${{ matrix.BASE_SOURCE_IMAGE_NAME }}
+        BASE_SOURCE_IMAGE_TAG: ${{ matrix.BASE_SOURCE_IMAGE_TAG }}
+        file: ${{ matrix.SOURCE_CONTEXT_DIR }}/${{ matrix.DOCKER_FILE_PATH }}
+        REGISTRY_USERNAME: ${{ secrets[matrix.REGISTRY_USERNAME_SECRET_NAME]}}
+        REGISTRY_PASSWORD: ${{ secrets[matrix.REGISTRY_PASSWORD_SECRET_NAME]}}
+
+    - name: Build and push Docker image
+      id: docker_build
+      if: contains(fromJSON('["aries-endorser-agent","aries-endorser-backup","aries-endorser-api","aries-endorser-proxy"]'), matrix.service)
+      uses: docker/build-push-action@v5
+      with:
+        context: ${{ matrix.SOURCE_CONTEXT_DIR }}
+        file: ${{ matrix.SOURCE_CONTEXT_DIR }}/${{ matrix.DOCKER_FILE_PATH }}
+        push: true
+        tags: ${{ steps.meta.outputs.tags }}
+        outputs: type=image,name=target,annotation-index.org.opencontainers.image.description=Hyperledger Aries Cloud Agent Python (ACA-Py) is a foundation for building decentralized id
+        labels: | 
+          io.openshift.build.source-location=${{ github.repositoryUrl }}
+          io.openshift.build.commit.id=${{ github.sha }}
+          
+    - name: Pull database image
+      if: contains(fromJSON('["aries-endorser-db"]'), matrix.service)
+      run: |
+        docker pull ${{ matrix.BUILDER_IMAGE }}
+    - name: Extract Values
+      id: extract
+      if: contains(fromJSON('["aries-endorser-db"]'), matrix.service)
+      run: echo "tags=$(echo '${{  steps.meta.outputs.tags  }}' | grep -oE ':([^[:space:]]+)' |  sed '/workflow/d' | sed 's/://g' | tr '\n' ' ')" >> $GITHUB_OUTPUT
+
+    - name: Build database image
+      id: build_image
+      if: contains(fromJSON('["aries-endorser-db"]'), matrix.service)
+      uses: redhat-actions/s2i-build@v2
+      with:
+        path_context: ${{ matrix.PATH }}
+        # Builder image for a java project
+        builder_image: ${{ matrix.BUILDER_IMAGE }}
+        image:  dts-endorser-service/${{ matrix.service }}
+        tags: ${{ steps.extract.outputs.tags }}
+
+    - name: Push database image
+      id: push
+      if: contains(fromJSON('["aries-endorser-db"]'), matrix.service)
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        tags: ${{ steps.build_image.outputs.tags }}
+        image:  ${{ steps.build_image.outputs.image }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+        registry: ghcr.io/bcgov
```

Comment:
You might be able to drop the `registry` parameter.  See note on the `redhat-actions/s2i-build` step.  Otherwise this should be `${{ env.GITHUB_IMAGE_REPO }}`.